docs/planning/netstandard-vnext/README.md

```diff
@@ -0,0 +1,127 @@
+
+This document describes the plan for .NET Standard *vNext*, which includes the
+definition of its API surface.
+
+## Goals
+
+* **Include `Span<T>`**. In .NET Core 2.1 we've added `Span<T>` which is an
+  array like type that allows representing managed and unmanaged memory in a
+  uniform way and supports cheap slicing. It's at the heart of most
+  performance-related improvements in .NET Core 2.1 and allows managing buffers
+  in a more efficient way as it helps in reducing allocations and copying.  
+  `Span<T>` is considered a core type and requires runtime and compiler support
+  in order to be fully leveraged. Please note that when we refer to `Span<T>` in
+  this document, we really mean the family of span-related features which
+  includes other types such as `ReadOnlySpan<T>`, `Memory<T>`,
+  `ReadOnlyMemory<T>`, and the new `System.Buffers` namespace.
+
+* **Include core-APIs working with spans**. While `Span<T>` is available as a
+  .NET Standard compatible NuGet package (`System.Memory`) adding this package
+  cannot extend the members of .NET Standard types that deal with spans. For
+  example, .NET Core 2.1 added many APIs that allow working with spans, such as
+  `Stream.Read(Span<Byte>)`. Part of the value proposition to add span to .NET
+  Standard is to add the companion APIs as those cannot be added using a NuGet
+  package.
+
+* **General Goodness**. Since .NET Core was open sourced, we've added many small
+  features across the base class libraries such as `System.HashCode` for
+  combining hash codes or new overloads on `System.String`. There are about ~800
+  new members in .NET Core that are on types that are already in .NET Standard
+  2.0.
+
+## Decisions
+
+We did a diff between .NET Core 2.1 and .NET Standard 2.0 to see which APIs that
+were added in .NET Core would be great candidates for inclusion in the standard.
+From that list we made the following decisions:
+
+* **Include** Brotli as `HttpClient` depends on it
+* **Not include** `Microsoft.CSharp` (the standard is language neutral)
+* **Not include** `Microsoft.VisualBasic` (the standard is language neutral)
+* **Not include** `DataAnnotations` (out-of-band today)
+* **Not include** `DiagnosticSource` (out-of-band today)
+* **Not include** `System.Collections.Immutable` (out-of-band today)
+* **Not include** `System.Reflection.Metadata` (out-of-band today)
+* **Not include** `System.Runtime.InteropServices.WindowsRuntime` (out-of-band
+  today)
+* **Not include** `System.Runtime.Loader` (currently a .NET Core only concept
+  and not ready for standardization)
+* **Not include** `RegexCompilationInfo` as it throws on .NET Core & Xamarin and
+  we don't plan on implementing the API ever.
+* **Not include** `System.Threading.Tasks.Dataflow` (out-of-band today)
+* **Not include** `IApplicationResourceStreamResolver` (was plumbing for
+  Silverlight convergence. It's obsoleted.)
+* **Not include** `System.Xml.Serialization.SchemaImporter` (doesn't seem
```

Comment:
What is the plan for `xsd.exe`? I've tried to generate XML schemas from netcore assemblies, and it failed.